lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -3045,6 +3056,20 @@ void WireOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
 
 std::optional<size_t> WireOp::getTargetResultIndex() { return 0; }
 
+LogicalResult WireOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  auto circuitOp = getOperation()->getParentOfType<CircuitOp>();
+  auto refType = dyn_cast<RefType>(getType(0));
```

Comment:
It's probably fine but hardcoding operand numbers is less readable and a maintenance burden.

Should this be `firrtl::dyn_cast` in case we ever support aliases to non-base?

Maybe this should recursively check the type for any references? Although post-LOA this does the trick, but making sure that's intentional.

Reply:
Oh, right, need to fix `Forceable` for non-base things in Wires for that API to make sense again (`getDataType()`), blargh.

Reply:
Right, `getDataType()` is approximately what I want here. However, I need the actual type, not the ref-stripped type.

I thought about the "recursive check" but opted not to as that only comes up for open aggregates and those are immediately factored out by LOA.

I switched over all the `dyn_cast` to use `type_dyn_cast` (which is what I think you were referring to?) in 63cd555df.

Reply:
Yeah, I mean `type_dyn_cast`, thanks.

We can make working with declarations and their storage/component types better separately, that's what `getDataType()` is supposed to be (it doesn't strip anything, it's as opposed to the type of the rwprobe result for Forceable's) but was borked and not yet fixed when Forceable's now can have things that aren't FIRRTLBaseType.